### PR TITLE
887 upgrade dependencies of libms

### PR DESCRIPTION
**Pull request by nichlaes**
Updates all dependabots suggestions

### CODE REVIEW

**Comment by prasadtalasila** [comment]
@nichlaes I have migrated the eslint config to javascript equivalent of it in this [PR](https://github.com/prasadtalasila/DTaaS-Public/pull/536). The actual file is [here](https://github.com/prasadtalasila/DTaaS-Public/blob/2519e4cbd17025dc1af14b1ab1d6c88628c26cc4/servers/execution/runner/eslint.config.js)

**Comment by prasadtalasila** [comment]
@nichlaes please see the [eslint config](https://github.com/INTO-CPS-Association/DTaaS/pull/945/files#diff-59bfa1b64c2fa9406dfa6a0f88a09fc9947ca4e4339ba8b16630f2915bd34de0) in PR #945 for another working stack

File: servers/lib/package.json [hunk 1]
```diff
@@ -10,72 +10,77 @@
...
     "build": "tsc",
     "clean": "npx rimraf build node_modules coverage dist src.svg test.svg",
     "format": "prettier --ignore-path ../.gitignore --write \"**/*.{ts,tsx,css,scss}\"",
     "graph": "npx madge --image src.svg src && npx madge --image test.svg test",
-    "start": "node dist/src/main.js",
-    "start:pm2": "pm2 start pm2.config.js",
+    "start": "tsc & node dist/src/main.js",
+    "start:pm2": "pm2 start pm2.config.cjs",
     "stop:pm2": "pm2 delete libms",
     "syntax": "eslint . --fix",
```
Comment (prasadtalasila) [inline servers/lib/package.json:23]: Gives the following error (only on Windows):

```bash
$ yarn syntax
yarn run v1.22.19
$ eslint . --fix

Oops! Something went wrong! :(

ESLint: 9.11.1

ConfigError: Config (unnamed): Unexpected key "0" found.
    at rethrowConfigError (C:\Users\user\git\DTaaS\servers\lib\node_modules\@eslint\config-array\dist\cjs\index.cjs:303:8)
    at C:\Users\user\git\DTaaS\servers\lib\node_modules\@eslint\config-array\dist\cjs\index.cjs:1098:5
    at Array.reduce (<anonymous>)
    at FlatConfigArray.getConfigWithStatus (C:\Users\user\git\DTaaS\servers\lib\node_modules\@eslint\config-array\dist\cjs\index.cjs:1091:43)
    at FlatConfigArray.getConfig (C:\Users\user\git\DTaaS\servers\lib\node_modules\@eslint\config-array\dist\cjs\index.cjs:1120:15)
    at C:\Users\user\git\DTaaS\servers\lib\node_modules\eslint\lib\eslint\eslint-helpers.js:354:54
    at C:\Users\user\git\DTaaS\servers\lib\node_modules\eslint\lib\eslint\eslint-helpers.js:296:32
    at Object.isAppliedFilter (C:\Users\user\git\DTaaS\servers\lib\node_modules\@nodelib\fs.walk\out\readers\common.js:12:31)
    at AsyncReader._handleEntry (C:\Users\user\git\DTaaS\servers\lib\node_modules\@nodelib\fs.walk\out\readers\async.js:86:20)
    at C:\Users\user\git\DTaaS\servers\lib\node_modules\@nodelib\fs.walk\out\readers\async.js:65:22
error Command failed with exit code 2.
```

File: servers/lib/test/e2e/app.e2e.spec.ts [hunk 2]
```diff
@@ -1,14 +1,10 @@
...
+import request from 'supertest';
 import fetch from 'cross-fetch';
-import {
-  ApolloClient,
-  DocumentNode,
-  HttpLink,
-  InMemoryCache,
-  gql,
-} from '@apollo/client';
+import { ApolloClient, DocumentNode, InMemoryCache, gql} from "@apollo/client/core/core.cjs";
```
Comment (prasadtalasila) [inline servers/lib/test/e2e/app.e2e.spec.ts:6]: can we import esm modules of the library with `mjs` extension?

File: servers/lib/tsconfig.json [hunk 3]
```diff
@@ -4,28 +4,35 @@
...
-    //"strictNullChecks": true,
-    //"strict": true, //enable all strict type-checking options
-    "target": "es2022",
+    "target": "ES2022",
     "types": ["node", "jest"], //use node and jest types
     "typeRoots": [
       "node_modules/@types" //use node_modules/@types for type definitions
-    ]
+    ],
+    "allowJs": true,
```
Comment (prasadtalasila) [inline servers/lib/tsconfig.json:34]: How does this setting help? Is it with imports inside tests? If the intention is to include JS code as well, please do add [checkJs](https://www.typescriptlang.org/tsconfig/#checkJs) setting as well.

**Review by prasadtalasila — COMMENTED** [review]
@nichlaes Thanks for the updates. Please see the comments.

File: servers/lib/test/e2e/app.e2e.spec.ts (hunk 2 above)
Comment (nichlaes) [reply servers/lib/test/e2e/app.e2e.spec.ts:6]: I had to change it, since @apollo client isn't esm yet

File: servers/lib/package.json (hunk 1 above)
Comment (nichlaes) [reply servers/lib/package.json:23]: I don't get that error

**Comment by nichlaes** [comment]
@prasadtalasila  Please have a look again. It should be good now.

**Pull request merged**